### PR TITLE
Overhaul of CLI arguments

### LINKED ISSUE #148: Inconsistent parameter loading from script arguments
I noticed recently that the z tip coordinate that is reported on the plotted images was for some reason offset by 0.5Å on the graphene example compared to what I was expecting it to be based on what I did in the OpenCL implementation. I looked into this more closely, and I think the issue is with the parameter loading from the plot script arguments.

The z distance is calculated by adding half of the amplitude to the scan z coordinate at: https://github.com/Probe-Particle/ppafm/blob/8763be1e7e851cb47a1294d49eb04fcd1c14b34b/ppafm/cli/plot_results.py#L239-L242 Notice that it uses the value from the params dict `PPU.params['Amplitude']`. These values are parsed from the command line arguments in the `apply_options` function, but it only accepts keywords that are already in the params dict: https://github.com/Probe-Particle/ppafm/blob/8763be1e7e851cb47a1294d49eb04fcd1c14b34b/ppafm/common.py#L228-L230

However, the command line argument for the amplitude is just `a`: https://github.com/Probe-Particle/ppafm/blob/8763be1e7e851cb47a1294d49eb04fcd1c14b34b/ppafm/cli/plot_results.py#L29 which does not match the value existing in the params dict: https://github.com/Probe-Particle/ppafm/blob/8763be1e7e851cb47a1294d49eb04fcd1c14b34b/ppafm/common.py#L48 so it never gets updated and it uses the default value which is different from the value supplied from the command line.

There are some other command line arguments that have the same issue. It would probably be beneficial to make some kind of dictionary of common command line arguments to use in the scripts to make sure that the names always match.

@ProkopHapala @ondrejkrejci @yakutovicha

### PR DESCRIPTION
**Pull request by NikoOinonen**
fixes #148 

Adds to `ppafm.common` a new class `CLIParser`, which holds a list of common commandline arguments that can be added to the scripts. E.g.
```python
parser = CLIParser()
parser.add_arguments(['output_format', 'Amplitude', 'charge', 'noPBC'])
```
The arguments are divided into two categories: those which directly correspond to one of the parameters in the `ppafm.common.params` dictionary, and others. The parameter arguments are checked so that their names match with the names in the parameter dictionary to avoid problems.

All the main CLI scripts are changed to use the new parser. But not all arguments are transferred to the pre-defined list; only the ones which are used in more than one place.

A couple of the arguments are renamed:
- `data_format` -> `output_format` to contrast with the added new argument `input_format`. @mondracek 
- `Bpower`-> `Bpauli` in order to be consistent with the parameter name in the params dictionary.
- `densityMayBeNegative` -> `no_negative_check` in `conv_rho.py` for a more descriptive name.

### CODE REVIEW

File: ppafm/common.py [hunk 1]
```diff
@@ -57,11 +58,174 @@
     'tip_base':  np.array( ['None', 0.00 ]),
     'Rtip'         :  30.0,
     'permit'       :  0.00552634959,
-    'Vrange':   0.0,
     'vdWDampKind' : 2,
     '#' : None
 }
 
+class CLIParser(ArgumentParser):
```
Comment (ProkopHapala) [inline ppafm/common.py:65]: Aha it is extension of ArgumentParser. Is it some common practice to make you own derived class of ArgumentParser?

**Review by ProkopHapala — APPROVED** [review]
I perhpas did not encounter the problem which this should solve, so I don;t fully see it it is worth it. But I believe you have good reason to do this.

I'm thinking about that "ArgumentParser" is some standard of-the-shelf python module which everybody knows and use. Now we modify it to ower own custom made solution, which may be a bit more confusing to people. But perhaps it is worth it.

File: ppafm/common.py (hunk 1 above)
Comment (NikoOinonen) [reply ppafm/common.py:65]: It's maybe not common practice, but I think a good compromise. The main concern for me was that the arguments were not consistent with the names in the `params` dictionary. I wrote about the example of the amplitude being read wrong in one of the scripts in the associated issue #148. If we add the arguments through this interface, it's less error prone, there is less repeated code, and the help messages are more consistent.

**Comment by NikoOinonen** [comment]
It's approved, but I'll wait for @mondracek to give his opinion, since the CLI arguments concern his other PR #179.

**Comment by mondracek** [comment]
Seems good to me. Just one thing: When running the example in ``ppafm/examples/PTCDA_Hartree_dz2``, the line
```ppafm-plot-results -k 0.5 -q -0.10 -a 2.0 2 --df -f npy```
in ``run.sh`` results in the following error:
```ppafm-plot-results: error: unrecognized arguments: 2```
As I understand it, this is not really an error introduced by the new code. Rather, the value ``2`` for the amplitude is given twice by mistake in the above line. The old way of handling the arguments probably failed to notice this mistake and tolerated the example script as it is. Or perhaps it did not and I just did not notice that the example was not working before. Anyway, I suggest @NikoOinonen that you correct this line in ``ppafm/examples/PTCDA_Hartree_dz2/run.sh`` and commit it to this branch. Then, let's proceed with merging to ``main``. Once this PR #185 is merged, I will go on with my issue #39.

**Comment by mondracek** [comment]
Okay, I think as this `` examples/PTCDA_Hartree_dz2/run.sh`` issue is literally one character in one file, I can as well repair it myself and push the commit. So I am going to do it now.

**Pull request merged**